File: src/net/java/sip/communicator/impl/protocol/sip/xcap/model/StringUtils.java
/*
 * SIP Communicator, the OpenSource Java VoIP and Instant Messaging client.
 *
 * Distributable under LGPL license.
 * See terms of license at gnu.org.
 */
package net.java.sip.communicator.impl.protocol.sip.xcap.model;

import java.io.*;

/**
 * Utility class that helps to work with <tt>String</tt> class.
 *
 * @author Grigorii Balutsel
 */
public final class StringUtils
{
    /**
     * This class cannot be instanced.
     */
    private StringUtils()
    {
    }

    /**
     * Indicates whether string is null or empty.
     *
     * @param s the string to analyze.
     * @return true if string is null or empty.
     */
    public static boolean isNullOrEmpty(String s)
    {
        return isNullOrEmpty(s, true);
    }

    /**
     * Indicates whether string is null or empty.
     *
     * @param s    the string to analyze.
     * @param trim indicates whether to trim the string.
     * @return true if string is null or empty.
     */
    public static boolean isNullOrEmpty(String s, boolean trim)
    {
        if (s == null)
        {
            return true;
        }
        if (trim)
        {
            s = s.trim();
        }
        return s.length() == 0;
    }

    /**
     * Indicates whether strings are equal.
     *
     * @param s1 the string to analyze.
     * @param s2 the string to analyze.
     * @return true if string are equal.
     */
    public static boolean isEquals(String s1, String s2)
    {
        return (s1 == null && s2 == null) || (s1 != null && s1.equals(s2));
    }

    /**
     * Creates <tt>InputStream</tt> from the string in UTF8 encoding.
     *
     * @param string the string to convert.
     * @return the <tt>InputStream</tt>.
     * @throws UnsupportedEncodingException if UTF8 is unsupported.
     */
    public static InputStream fromString(String string)
            throws UnsupportedEncodingException
    {
        return fromString(string, "UTF-8");
    }

    /**
     * Creates <tt>InputStream</tt> from the string in the specified encoding.
     *
     * @param string   the string to convert.
     * @param encoding the encoding
     * @return the <tt>InputStream</tt>.
     * @throws UnsupportedEncodingException if the encoding is unsupported.
     */
    public static InputStream fromString(String string, String encoding)
            throws UnsupportedEncodingException
    {
        byte[] bytes = string.getBytes(encoding);
        return new ByteArrayInputStream(bytes);
    }
}

File: src/native/windows/msofficecomm/StringUtils.cxx
/*
 * Jitsi, the OpenSource Java VoIP and Instant Messaging client.
 *
 * Distributable under LGPL license.
 * See terms of license at gnu.org.
 */
#include "StringUtils.h"

LPWSTR StringUtils::MultiByteToWideChar(LPCSTR str)
{
    int wsize = ::MultiByteToWideChar(CP_ACP, 0, str, -1, NULL, 0);
    LPWSTR wstr;

    if (wsize)
    {
        wstr = (LPWSTR) ::malloc(wsize * sizeof(WCHAR));
        if (str && !::MultiByteToWideChar(CP_ACP, 0, str, -1, wstr, wsize))
        {
            ::free(wstr);
            wstr = NULL;
        }
    }
    else
        wstr = NULL;
    return wstr;
}

LPSTR StringUtils::WideCharToMultiByte(LPCWSTR wstr)
{
    int size = ::WideCharToMultiByte(CP_ACP, 0, wstr, -1, NULL, 0, NULL, NULL);
    LPSTR str;

    if (size)
    {
        str = (LPSTR) ::malloc(size);
        if (str
                && !::WideCharToMultiByte(
                        CP_ACP,
                        0,
                        wstr,
                        -1,
                        str,
                        size,
                        NULL,
                        NULL))
        {
            ::free(str);
            str = NULL;
        }
    }
    else
        str = NULL;
    return str;
}
